lib/util/git/index.ts

```diff
@@ -325,14 +328,16 @@ export async function branchExists(branchName: string): Promise<boolean> {
 
 // Return the commit SHA for a branch
 export async function getBranchCommit(branchName: string): Promise<CommitSha> {
-  await syncGit();
-  if (!(await branchExists(branchName))) {
-    throw Error(
-      'Cannot fetch commit for branch that does not exist: ' + branchName
-    );
+  if (!Object.keys(config.branchCommits).length) {
```

Comment:
We assume there must always be 1 or more branches in a remote

Reply:
Move this to init, so we can use it for baseBranches modified check too?

Then we are able to skip deps scan while have multiple base branches.

🤔

Reply:
Good point, it's hopefully little overhead to do the ls-remote every time

Reply:
I think so, because most git hoster will cache those Infos. I've read about this on latest bitbucket server changes some some ago.